addons/csw/functions/fnc_compatibleMagazines.sqf

```diff
@@ -0,0 +1,47 @@
+#include "..\script_component.hpp"
+/*
+ * Author: LinkIsGrim
+ * Gets all carry magazines that can be loaded into a CSW, includes weapons added by script.
+ *
+ * Arguments:
+ * 0: CSW <OBJECT>
+ *
+ * Return Value:
+ * Compatible Magazines <HASHMAP>
+ *   Magazine classname <STRING>
+ *   Nothing
+ *
+ * Example:
+ * [cursorObject] call ace_csw_fnc_compatibleMagazines
+ *
+ * Public: Yes
+ */
+
+params [["_csw", objNull, [objNull]]];
+
+if !((typeOf _csw) in GVAR(initializedStaticTypes)) exitWith {createHashMap};
+
+// fast exit for csw with single weapon, most common scenario
+if (count allTurrets _csw isEqualTo 1 && {count weapons _csw isEqualTo 1}) exitWith {
+    +(GVAR(compatibleMagsCache) get ((weapons _csw) select 0)) // return
```

Comment:
If `defaultAssemblyMode` is set to `false`, but `ammoHandling` is set to `2`, it throws an error because the return is `nil`. The CBA settings do not make sense, but atm it's possible to have them set as such.

```suggestion
    +(GVAR(compatibleMagsCache) getOrDefault [(weapons _csw) select 0, createHashMap]) // return
```